KinWallet/View Controllers and Co./More/HelpCenterViewController.swift

```diff
@@ -87,43 +133,101 @@ private class HelpCenterWebViewHandler: NSObject, WKScriptMessageHandler {
+            handleSubmissionError(body: body)
         case MessageNames.contactSupport:
-            helpCenterViewController?.contactSupport(category: category, pageTitle: title)
+            contactSupportClicked(body: body)
         case MessageNames.isPageHelpfulSelection:
-            if let isHelpful = body["helpful"] as? Bool {
-                isPageHelpfulSelection(category: category, pageTitle: title, isHelpful: isHelpful)
-            }
+            isPageHelpfulSelection(body: body)
         case MessageNames.pageLoaded:
-            if category == "Main Page" {
-                mainPageLoaded()
-            } else {
-                pageLoaded(category: category, pageTitle: title)
-            }
+            pageLoaded(body: body)
+        case MessageNames.supportSubmitted:
+            supportSubmitted(body: body)
+        case MessageNames.supportRequestSent:
+            supportRequestSent(body: body)
+        case MessageNames.formPageLoaded:
+            formPageLoaded(body: body)
         default:
             break
         }
     }
 
-    private func mainPageLoaded() {
-        Events.Analytics.ViewFaqMainPage().send()
+    private func handleSubmissionError(body: [String: Any]) {
+        guard
+            let data = body["data"] as? String,
+            let errors = body["error"] as? String else {
+                return
+        }
+        print("Data: \(data), Error: \(errors)")
+        helpCenterViewController?.presentErrorAlert()
+    }
+
+    private func contactSupportClicked(body: [String: Any]) {
+        guard
+            let category = body[FAQKeys.category] as? String,
+            let subCategory = body[FAQKeys.subCategory] as? String else {
+                return
+        }
+        Events.Analytics
+            .ClickContactButtonOnSpecificFaqPage(faqCategory: category, faqSubcategory: subCategory)
+            .send()
+    }
+
+    private func pageLoaded(body: [String: Any]) {
+        guard
+            let category = body[FAQKeys.category] as? String,
+            let subCategory = body[FAQKeys.subCategory] as? String else {
+                return
+        }
+        if category == "FAQ" {
+            Events.Analytics.ViewFaqMainPage().send()
+        } else {
+            Events.Analytics
+                .ViewFaqPage(faqCategory: category, faqSubcategory: subCategory)
+                .send()
+        }
     }
 
-    private func pageLoaded(category: String, pageTitle: String) {
+    private func isPageHelpfulSelection(body: [String: Any]) {
+        guard
+            let isHelpful = body["helpful"] as? Bool,
+            let category = body[FAQKeys.category] as? String,
+            let subCategory = body[FAQKeys.subCategory] as? String else {
+                return
+        }
         Events.Analytics
-            .ViewFaqPage(faqCategory: category, faqTitle: pageTitle)
+            .ClickPageHelpfulButtonOnFaqPage(faqCategory: category, faqSubcategory: subCategory, helpful: isHelpful)
```

Comment:
Doesn't this line trigger the 120 characters warning?

Reply:
no